### PR TITLE
Managed cassandra preview commands for cli extension

### PR DESCRIPTION
**Pull request by vivek-microsoft**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by evelyn-ys** [comment]
Please add tests for new commands

**Comment by evelyn-ys** [comment]
Since the api version has upgraded, I think you might need to re-record all the existing tests.

**Comment by jiasli** [comment]
Please sync from `main` to avoid pipeline being run for other extensions.

File: src/cosmosdb-preview/azext_cosmosdb_preview/custom.py [hunk 1]
```diff
@@ -170,29 +234,76 @@ def cli_cosmosdb_managed_cassandra_datacenter_create(client,
...
+        sku=sku,
+        disk_sku=disk_sku,
+        disk_capacity=disk_capacity,
+        availability_zone=availability_zone,
+        managed_disk_customer_key_uri=managed_disk_customer_key_uri,
+        backup_storage_customer_key_uri=backup_storage_customer_key_uri
     )
 
+    if isAllNone is False:
+        data_center_properties.authentication_method_ldap_properties=ldap_properties
```
Comment (vivek-microsoft) [inline src/cosmosdb-preview/azext_cosmosdb_preview/custom.py:283]: When this if condition is not met, data_center_properties.authentication_method_ldap_properties is not set at all, but still the serialized payload contains data_center_properties.authentication_method_ldap_properties as {}

Instead I expect authentication_method_ldap_properties key not to be part of the payload itself.

File: src/cosmosdb-preview/azext_cosmosdb_preview/commands.py [hunk 2]
```diff
@@ -34,15 +31,20 @@ def load_command_table(self, _):
...
+    with self.command_group('managed-cassandra cluster', cosmosdb_managed_cassandra_cluster_sdk, client_factory=cf_cassandra_cluster) as g:
         g.custom_command('create', 'cli_cosmosdb_managed_cassandra_cluster_create', supports_no_wait=True)
         g.custom_command('update', 'cli_cosmosdb_managed_cassandra_cluster_update', supports_no_wait=True)
-        g.custom_command('node-status', 'cli_cosmosdb_managed_cassandra_fetch_node_status', table_transformer=amc_node_status_table_format, supports_no_wait=True)
+        g.custom_command('backup list', 'cli_cosmosdb_managed_cassandra_cluster_list_backup', is_preview=True)
+        g.custom_command('backup show', 'cli_cosmosdb_managed_cassandra_cluster_show_backup', is_preview=True)
+        g.custom_command('deallocate', 'cli_cosmosdb_managed_cassandra_cluster_deallocate', supports_no_wait=True)
+        g.custom_command('invoke-command', 'cli_cosmosdb_managed_cassandra_cluster_invoke_command', supports_no_wait=True)
+        g.custom_command('start', 'cli_cosmosdb_managed_cassandra_cluster_start', supports_no_wait=True)
+        g.custom_command('status', 'cli_cosmosdb_managed_cassandra_cluster_status')
```
Comment (evelyn-ys) [inline src/cosmosdb-preview/azext_cosmosdb_preview/commands.py:42]: These commands have already been in main CLI, why add them in extension again?

File: src/cosmosdb-preview/azext_cosmosdb_preview/_help.py [hunk 3]
```diff
@@ -72,16 +72,31 @@
       az managed-cassandra cluster list
 """
 
-helps['managed-cassandra cluster node-status'] = """
+helps['managed-cassandra cluster backup-list'] = """
```
Comment (evelyn-ys) [inline src/cosmosdb-preview/azext_cosmosdb_preview/_help.py:75]: Generally we will design CLI commands like `managed0cassandra cluster backup list`

File: src/cosmosdb-preview/azext_cosmosdb_preview/_help.py [hunk 4]
```diff
@@ -72,16 +72,31 @@
...
 examples:
-  - name: This command gets the status of all the nodes in this cluster. By default a json is returned.
+  - name: This command lists the backups of this cluster that are available to restore.
     text: |
-      az managed-cassandra cluster node-status --resource-group MyResourceGroup --cluster-name MyCluster
-  - name: This command gets the status of all the nodes in this cluster. When a table output is specified only one token is displayed. Use json output to get all the tokens.
+      az managed-cassandra cluster backup-list --resource-group MyResourceGroup --cluster-name MyCluster
+"""
+
+helps['managed-cassandra cluster backup-show'] = """
```
Comment (evelyn-ys) [inline src/cosmosdb-preview/azext_cosmosdb_preview/_help.py:84]: Same here `managed-cassandra cluster backup show`

File: src/cosmosdb-preview/azext_cosmosdb_preview/_params.py [hunk 5]
```diff
@@ -8,20 +8,24 @@
...
+    from azure.cli.core.commands.parameters import tags_type, get_enum_type, get_three_state_flag
 
     # Managed Cassandra Cluster
     for scope in [
             'managed-cassandra cluster create',
             'managed-cassandra cluster update',
             'managed-cassandra cluster show',
             'managed-cassandra cluster delete',
-            'managed-cassandra cluster node-status']:
+            'managed-cassandra cluster backup-list']:
```
Comment (evelyn-ys) [inline src/cosmosdb-preview/azext_cosmosdb_preview/_params.py:28]:
```suggestion
            'managed-cassandra cluster backup-list',
            'managed-cassandra cluster backup-show']:
```

File: src/cosmosdb-preview/azext_cosmosdb_preview/commands.py [hunk 6]
```diff
@@ -34,15 +31,16 @@ def load_command_table(self, _):
         operations_tmpl='azext_cosmosdb_preview.vendored_sdks.azure_mgmt_cosmosdb.operations#CassandraDataCentersOperations.{}',
         client_factory=cf_cassandra_data_center)
 
-    with self.command_group('managed-cassandra cluster', cosmosdb_managed_cassandra_cluster_sdk, client_factory=cf_cassandra_cluster, is_preview=True) as g:
+    with self.command_group('managed-cassandra cluster', cosmosdb_managed_cassandra_cluster_sdk, client_factory=cf_cassandra_cluster) as g:
         g.custom_command('create', 'cli_cosmosdb_managed_cassandra_cluster_create', supports_no_wait=True)
         g.custom_command('update', 'cli_cosmosdb_managed_cassandra_cluster_update', supports_no_wait=True)
-        g.custom_command('node-status', 'cli_cosmosdb_managed_cassandra_fetch_node_status', table_transformer=amc_node_status_table_format, supports_no_wait=True)
+        g.custom_command('backup-list', 'cli_cosmosdb_managed_cassandra_cluster_list_backup', is_preview=True)
+        g.custom_command('backup-show', 'cli_cosmosdb_managed_cassandra_cluster_show_backup', is_preview=True)
```
Comment (evelyn-ys) [inline src/cosmosdb-preview/azext_cosmosdb_preview/commands.py:38]: Better to change them to `backup list` and `backup show` like all other commands.

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**